buttervolume/plugin.py

```diff
@@ -271,3 +271,63 @@ def snapshot_restore():
     else:
         res['Err'] = 'No such snapshot'
     return json.dumps(res)
+
+
+@route('/VolumeDriver.Snapshots.Purge', ['POST'])
+def snapshots_purge():
+    """
+    Purge snapshots with a retention pattern
+    (see cli help)
+    """
+    params = jsonloads(request.body.read())
+    volume_name = params['Name']
+    dryrun = params.get('Dryrun', False)
+    units = {'m': 1, 'h': 60, 'd': 60*24, 'w': 60*24*7, 'y': 60*24*365}
+    try:
+        pattern = sorted(int(i[:-1])*units[i[-1]]
+                         for i in params['Pattern'].split(':'))
+        assert(len(pattern) >= 2)
+        max_age = pattern[-1]
+    except:
+        return {'Err': 'Invalid purge pattern'}
+    # snapshots related to the volume, more recents first
+    snapshots = sorted([s for s in os.listdir(SNAPSHOTS_PATH)
```

Comment:
not sure with python 3.5 I through sorted do not return a new list but change the list in place